src/reducer/index.js

```diff
@@ -0,0 +1,65 @@
+const initialState = {
+  restaurants: [],
+  regions: [],
+  categories: [],
+  search: {
+    region: '',
+    categoryId: '',
+  },
+};
+
+export default function reducer(state = initialState, action) {
+  const { type } = action;
+
+  function setRegions() {
+    const { regions } = action.payload;
+
+    return {
+      ...state,
+      regions,
+    };
+  }
+
+  function setCategories() {
+    const { categories } = action.payload;
+
+    return {
+      ...state,
+      categories,
+    };
+  }
+
+  function setRestaurants() {
+    const { restaurants } = action.payload;
+
+    return {
+      ...state,
+      restaurants,
+    };
+  }
+
+  function changeSearch() {
+    const { search, value } = action.payload;
+
+    return {
+      ...state,
+      search: {
+        ...state.search,
+        [search]: value,
+      },
+    };
+  }
+
+  const actionType = {
+    setRegions,
+    setCategories,
+    setRestaurants,
+    changeSearch,
+  };
+
+  if (!actionType[type]) {
+    return state;
+  }
+
+  return actionType[type]();
```

Comment:
```suggestion
  function defaultAction() {
    return state;
  }

  return (actionType[type] || defaultAction)();
```

위와 같이 액션 타입이 없는 경우에 대해 함수로 처리 해줄 수도 있겠네요~

Reply:
피드백 감사합니다. https://github.com/CodeSoom/react-week5-assignment-1/pull/59/commits/f154bb74cf572591f5f94f3892834f51037aec97 에서 변경해주었습니다!